### ISSUE TITLE
[5.5.1] New Doorkeper version breaks assertion strategy

### ISSUE THREAD

**Issue by jpalumickas** [doorkeeper-gem/doorkeeper#1498]
### Steps to reproduce

In version 5.5.0 was working everything fine, but after upgrading to v5.5.1 when using assertion strategy (`doorkeeper-grants_assertion` gem) token is creating with `resource_owner_id` `nil` Downgrading to `5.5.0` fixes the problem.
```rb
  resource_owner_from_assertion do
    User.find(...)
  end
```

### Expected behavior
It should set `resource_owner_id` as expected

### Actual behavior
`resource_owner_id` is `nil`

### System configuration
You can help us to understand your problem if you will share some very
useful information about your project environment (don't forget to
remove any confidential data if it exists).

**Doorkeeper initializer**:

```ruby
# config/initializers/doorkeeper.rb
Doorkeeper.configure do
Doorkeeper.configure do
  orm :active_record

  resource_owner_authenticator do
    raise "Please configure doorkeeper resource_owner_authenticator block located in #{__FILE__}"
  end

  api_only

  use_refresh_token

  access_token_methods :from_bearer_authorization, :from_access_token_param

  grant_flows(%w[authorization_code client_credentials password assertion])

  realm 'My Realm'

  resource_owner_from_credentials do
    # find and return user by using params
  end

  resource_owner_from_assertion do
    # find and return user by using params
  end
end
```

**Ruby version**: `2.7.2`

**Comment by nbulaj** [comment]
Hi @jpalumickas . Actually I'm surprised that `doorkeeper-grants_assertion` still works with the gem.

But other thing that surprises me is that not so much changes were introduced in 5.5.1: https://github.com/doorkeeper-gem/doorkeeper/compare/v5.5.0..v5.5.1 :thinking:

**Comment by nbulaj** [comment]
Ah, OK, so this what breaks https://github.com/doorkeeper-gem/doorkeeper/compare/v5.5.0..v5.5.1#diff-2e4651a4efa020d6b628f9bb3f7add09229e60030d842768e361d593fab4b6dbR15

**Comment by nbulaj** [comment]
Fixed in the extension and released as 0.3.1

Anyway I don't support this extension, so no guarantee for it :disappointed:

**Comment by jpalumickas** [comment]
@nbulaj thanks! it's working now 🙂
